imqsauth.go

```diff
@@ -234,6 +251,221 @@ func createDB(config *authaus.Config) bool {
 	return true
 }
 
+func healthCheck(ic *auth.ImqsCentral, cleanAll bool) bool {
+	db := ic.Central.GetRoleGroupDB()
+	sb := strings.Builder{}
+	w := tabwriter.NewWriter(&sb, 0, 8, 1, '\t', tabwriter.AlignRight)
+	errorSet := map[string]error{}
+	_, err := fmt.Fprintf(w, "\n")
+	if err != nil {
+		// do nothing
+	}
+	fmt.Fprintf(w, "Health Check\n")
+	fmt.Fprintf(w, "------------\n")
+
+	// General info
+	// ------------
+
+	// Migration version
+	v, e := authaus.GetMigrationVersion(&ic.Config.Authaus.DB)
+	if e != nil {
+		fmt.Fprintf(w, "Could not read migration version...\n%v\n", e)
+	} else {
+		fmt.Fprintf(w, "Migration version:\t%v\n", v)
+	}
+
+	// Groups
+	groups, err := db.GetGroups()
+	if err == nil {
+		fmt.Fprintf(w, "Total nr of groups:\t%v\n", len(groups))
+	} else {
+		fmt.Fprintf(w, "Total nr of groups:\t%v\n", err)
+	}
+
+	// Users
+	users, err := ic.Central.GetAuthenticatorIdentities(authaus.GetIdentitiesFlagDeleted)
+	if err == nil {
+		fmt.Fprintf(w, "Total nr of users:\t%v\n", len(users))
+	} else {
+		fmt.Fprintf(w, "Total nr of users:\t%v\n", err)
+	}
+
+	// Non-archived users without sessions
+	tokens, e := ic.Central.GetAllTokens(false)
+	if e != nil {
+		fmt.Fprintf(w, "Could not retrieve tokens from session db\n")
+	} else {
+		fmt.Fprintf(w, "Active auth session tokens retrieved:\t%v\n", len(tokens))
+	}
+
+	oAuthTokenIDs, e := ic.Central.GetAllOAuthTokenIDs()
+	if e != nil {
+		fmt.Fprintf(w, "Could not retrieve tokens from session db\n")
+	} else {
+		fmt.Fprintf(w, "OAuth session tokens retrieved:\t%v\n", len(oAuthTokenIDs))
+	}
+
+	// detect any missing oauth sessions
+	nrSessionsWithOrphanedOAuthIds := findMissingOAuthTokens(oAuthTokenIDs, tokens)
+	fmt.Fprintf(w, "Nr of sessions with orphaned OAuth IDs:\t%v\n", nrSessionsWithOrphanedOAuthIds)
```

Comment:
I would only call this if `GetAllOAuthTokenIDs` did not return an error, as currently you might run this function with a `nil` as the first argument :(

Reply:
Oops